tapdb/universe_forest.go

```diff
@@ -125,3 +130,69 @@ func (b *BaseUniverseForest) RootNodes(
 
 	return uniRoots, nil
 }
+
+// RegisterIssuance inserts a new minting leaf within the universe tree, stored
+// at the base key.
+func (b *BaseUniverseForest) RegisterIssuance(ctx context.Context,
+	id universe.Identifier, key universe.BaseKey,
+	leaf *universe.MintingLeaf,
+	metaReveal *proof.MetaReveal) (*universe.IssuanceProof, error) {
+
+	var (
+		writeTx       BaseUniverseForestOptions
+		issuanceProof *universe.IssuanceProof
+	)
+
+	dbErr := b.db.ExecTx(
+		ctx, &writeTx, func(dbTx BaseUniverseForestStore) error {
+			// Register issuance in the asset (group) specific
+			// universe tree.
+			var (
+				universeRoot mssmt.Node
+				err          error
+			)
+			issuanceProof, universeRoot, err = universeRegisterIssuance(
+				ctx, dbTx, id, key, leaf, metaReveal,
+			)
+			if err != nil {
+				return err
+			}
+
+			// Retrieve a handle to the multiverse tree so that we
+			// can update the tree by inserting a new issuance.
+			multiverseTree := mssmt.NewCompactedTree(
+				newTreeStoreWrapperTx(dbTx, mssmtNamespace),
+			)
+
+			// Construct a leaf node for insertion into the
+			// multiverse tree. The leaf node includes a reference
+			// to the lower tree via the lower tree root hash.
+			universeRootHash := universeRoot.NodeHash()
+			assetGroupSum := universeRoot.NodeSum()
+
+			leafNode := mssmt.NewLeafNode(
+				universeRootHash[:], assetGroupSum,
+			)
+
+			// Use asset ID (or asset group hash) as the upper tree
+			// leaf node key. This is the same as the asset specific
+			// universe ID.
+			leafNodeKey := id.Bytes()
+
+			_, err = multiverseTree.Insert(
```

Comment:
The issuance proof returned here should now also include the inclusion path into the multi-verse root.